x-pack/plugins/apm/server/index.ts

```diff
@@ -48,11 +47,44 @@ const configSchema = schema.object({
       enabled: schema.boolean({ defaultValue: false }),
     }),
   }),
+  transactionIndices: schema.string({
+    defaultValue: 'traces-apm*,apm-*-transaction-*',
+  }),
+  spanIndices: schema.string({ defaultValue: 'traces-apm*,apm-*-span-*' }),
+  errorIndices: schema.string({ defaultValue: 'logs-apm*,apm-*-error-*' }),
+  metricsIndices: schema.string({
+    defaultValue: 'metrics-apm*,apm-*-metric-*',
+  }),
+  sourcemapIndices: schema.string({ defaultValue: 'apm-*-sourcemap-*' }),
+  onboardingIndices: schema.string({ defaultValue: 'apm-*-onboarding-*' }),
+  indexPattern: schema.string({ defaultValue: 'apm-*' }),
+  fleetMode: schema.boolean({ defaultValue: true }),
 });
 
 // plugin config
 export const config: PluginConfigDescriptor<APMXPackConfig> = {
-  deprecations: ({ deprecate }) => [deprecate('enabled', '8.0.0')],
+  deprecations: ({
+    deprecate,
+    renameFromRoot,
+    deprecateFromRoot,
+    unusedFromRoot,
+  }) => [
+    deprecate('enabled', '8.0.0'),
+    renameFromRoot(
+      'apm_oss.transactionIndices',
+      'xpack.apm.transactionIndices'
+    ),
+    renameFromRoot('apm_oss.spanIndices', 'xpack.apm.spanIndices'),
+    renameFromRoot('apm_oss.errorIndices', 'xpack.apm.errorIndices'),
+    renameFromRoot('apm_oss.metricsIndices', 'xpack.apm.metricsIndices'),
+    renameFromRoot('apm_oss.sourcemapIndices', 'xpack.apm.sourcemapIndices'),
+    renameFromRoot('apm_oss.onboardingIndices', 'xpack.apm.onboardingIndices'),
+    renameFromRoot('apm_oss.indexPattern', 'xpack.apm.indexPattern'),
+    renameFromRoot('apm_oss.fleetMode', 'xpack.apm.fleetMode'),
+    deprecateFromRoot('apm_oss.enabled', '8.0.0'),
+    unusedFromRoot('apm_oss.fleetMode'),
+    unusedFromRoot('apm_oss.indexPattern'),
```

Comment:
Would it make sense to fix https://github.com/elastic/kibana/issues/96278 while you are at it?